File: src/main/java/org/lamisplus/modules/prep/util/EnrollmentType.java
package org.lamisplus.modules.prep.util;

/**
 * Canonical codes stored in {@code prophylaxis_initiation.enrollment_type}.
 * These are the codeset values from PREP_PEP_ENROLLMENT_TYPE — never compare
 * against "PrEP" / "PEP" string literals; use these constants instead.
 *
 * <p>The constants are {@code public static final String} (not an enum) so they
 * can be inlined into JPA {@code @Query} native SQL via string concatenation.
 */
public final class EnrollmentType {
    public static final String PREP = "PREP_PEP_ENROLLMENT_TYPE_PREP";
    public static final String PEP = "PREP_PEP_ENROLLMENT_TYPE_PEP";

    public static boolean isPrep(String code) {
        return PREP.equalsIgnoreCase(code);
    }

    public static boolean isPep(String code) {
        return PEP.equalsIgnoreCase(code);
    }

    /** Maps a UI label ("PrEP" / "PEP") or canonical code to the canonical code. */
    public static String toCanonical(String value) {
        if (value == null) return null;
        if ("PrEP".equalsIgnoreCase(value) || PREP.equalsIgnoreCase(value)) return PREP;
        if ("PEP".equalsIgnoreCase(value) || PEP.equalsIgnoreCase(value)) return PEP;
        return value;
    }

    private EnrollmentType() {}
}

File: src/main/webapp/jsx/constants/enrollmentType.js
// Canonical codeset codes stored in prophylaxis_initiation.enrollment_type.
// Never send the short labels ("PrEP" / "PEP") to the backend — use these
// constants. UI display can still use the short labels.

export const ENROLLMENT_TYPE_PREP = "PREP_PEP_ENROLLMENT_TYPE_PREP";
export const ENROLLMENT_TYPE_PEP = "PREP_PEP_ENROLLMENT_TYPE_PEP";

export const ENROLLMENT_LABEL_PREP = "PrEP";
export const ENROLLMENT_LABEL_PEP = "PEP";

// Convert a UI label or already-canonical value to the canonical code.
export const toEnrollmentTypeCode = (value) => {
  if (!value) return value;
  if (value === ENROLLMENT_LABEL_PREP || value === ENROLLMENT_TYPE_PREP) {
    return ENROLLMENT_TYPE_PREP;
  }
  if (value === ENROLLMENT_LABEL_PEP || value === ENROLLMENT_TYPE_PEP) {
    return ENROLLMENT_TYPE_PEP;
  }
  return value;
};

// Convert a canonical code back to the short UI label.
export const fromEnrollmentTypeCode = (value) => {
  if (value === ENROLLMENT_TYPE_PREP) return ENROLLMENT_LABEL_PREP;
  if (value === ENROLLMENT_TYPE_PEP) return ENROLLMENT_LABEL_PEP;
  return value;
};

export const isPrep = (value) =>
  value === ENROLLMENT_TYPE_PREP || value === ENROLLMENT_LABEL_PREP;

export const isPep = (value) =>
  value === ENROLLMENT_TYPE_PEP || value === ENROLLMENT_LABEL_PEP;
